tests/phpunit/tests/user/capabilities.php

```diff
@@ -980,6 +980,461 @@ public function test_add_role() {
 		$this->assertFalse( $wp_roles->is_role( $role_name ) );
 	}
 
+	/**
+	 * @dataProvider data_update_role_unhappy_paths
+	 *
+	 * @ticket 54572
+	 *
+	 * @covers WP_Roles::update_role
+	 * @covers ::update_role
+	 *
+	 * @param mixed $role         The role to update.
+	 * @param mixed $display_name The display name for the role.
+	 * @param mixed $capabilities The capabilities for the role.
+	 */
+	public function test_update_role_should_not_try_to_update_a_role( $role, $display_name, $capabilities ) {
+		global $wp_roles;
+
+		// Create role if it does not exist.
+		$role_name     = 'janitor';
+		$expected_caps = array(
+			'edit_posts' => true,
+			'edit_pages' => true,
+			'level_0'    => true,
+			'level_1'    => true,
+			'level_2'    => true,
+		);
+		add_role( $role_name, 'Janitor', $expected_caps );
+		$this->flush_roles();
+
+		$this->assertTrue(
+			$wp_roles->is_role( $role_name ),
+			"The $role_name role was not created"
+		);
+
+		$this->assertNotInstanceOf(
+			'WP_Roles',
+			update_role( $role_name, 'Janitor', $expected_caps ),
```

Comment:
```php
			update_role( $role_name, 'Janitor', $expected_caps ),
```

This is why, it's using the incorrect variable names -- it should be using the ones passed by the data provider.